cmd/state-offline-installer/install.go

```diff
@@ -60,17 +63,46 @@ func NewRunner(prime primeable) *runner {
 }
 
 type Params struct {
-	path string
+	path   string
+	config *InstallerConfig
 }
 
-func newParams() *Params {
-	return &Params{path: "/tmp"}
+func newParams(config *InstallerConfig) *Params {
+	return &Params{
+		path:   "/tmp",
+		config: config,
+	}
+}
+
+func (r *runner) Event(eventType string, installerDimensions *dimensions.Values) {
+	go r.analytics.Event(ac.CatRuntimeUsage, eventType, installerDimensions)
+}
+
+func (r *runner) EventWithLabel(eventType string, msg string, installerDimensions *dimensions.Values) {
+	go r.analytics.EventWithLabel(ac.CatRuntimeUsage, eventType, msg, installerDimensions)
+}
+
+func (r *runner) handleFailure(err error, msg string, installerDimensions *dimensions.Values) error {
+	r.EventWithLabel("failure", msg, installerDimensions)
```

Comment:
Do you want to do this one in a goroutine as well?

Reply:
That would be a good idea ;)